ext/standard/streamsfuncs.c

```diff
@@ -1774,6 +1774,46 @@ PHP_FUNCTION(sapi_windows_vt100_support)
 		}
 	}
 }
+
+PHP_FUNCTION(sapi_windows_console_size)
+{
+	zval *zsrc;
+	php_stream *stream;
+	zend_long fileno;
+	int width, height;
+	zval zwidth, zheight;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+	Z_PARAM_RESOURCE(zsrc)
+	ZEND_PARSE_PARAMETERS_END();
+
+	php_stream_from_zval(stream, zsrc);
+
+	/* get the fd.
+	 * NB: Most other code will NOT use the PHP_STREAM_CAST_INTERNAL flag when casting.
+	 * It is only used here so that the buffered data warning is not displayed.
+	 */
+	if (php_stream_can_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL) == SUCCESS) {
+		php_stream_cast(stream, PHP_STREAM_AS_FD_FOR_SELECT | PHP_STREAM_CAST_INTERNAL, (void*)&fileno, 0);
+	} else if (php_stream_can_cast(stream, PHP_STREAM_AS_FD | PHP_STREAM_CAST_INTERNAL) == SUCCESS) {
+		php_stream_cast(stream, PHP_STREAM_AS_FD | PHP_STREAM_CAST_INTERNAL, (void*)&fileno, 0);
+	} else {
+		php_error_docref(NULL, E_WARNING, "not able to analyze the specified stream");
```

Comment:
After reading https://bugs.php.net/bug.php?id=79805, I think the warning should be dropped (we only warn about this when `sapi_windows_vt100_support()` is used as setter).